integration_tests/src/main/python/data_gen.py

```diff
@@ -144,9 +154,9 @@ def __init__(self, child_gen, func, data_type=None, nullable=True):
         super().__init__(data_type, nullable=nullable)
         self._child_gen = child_gen
         self._func = func
-
+    
     def __repr__(self):
-        return super().__repr__() + '(' + str(self._child_gen) + ')'
+        return super().__repr__() + '(' + self._child_gen.__repr__() + ')'
 
```

Comment:
No, but they need to override `__repr__`. `__repr__` will be part of the test case name, so if `specialcases` are placed there, the test case name will be very long and unreadable, so I used `cache_repr` to store them.

Reply:
Yes, It was a mistake, thank you! They need to override it because the nested types may also have their special cases. I removed the `cache_repr` now and hashed the `specialcases` to shorten the `__repr__`.